File: server/status.py
import requests

def ed_status() -> str:
    resp = requests.get("https://ed-server-status.orerve.net")
    if resp.ok:
        json_data = resp.json()
        return json_data['status']
    else:
        return 'Unknown Status Error'

def set_status(message: str, colour: str) -> None:
    if colour == "green":
        emoji = "🟢"
    elif colour == "yellow":
        emoji = "🟡"
    else:
        emoji = "🔴"
    try:
        open("cache/status.txt", "w").write(f"{message}\n{emoji}")
    except Exception:
        open("cache/status.txt", "x").write(f"{message}\n{colour}")
    return

def get_status() -> list[str]:
    try:
        data = open("cache/status.txt", "r").readlines()
        return [data[0], data[1], ed_status()]
    except Exception:
        open("cache/status.txt", "x").write(f"All Operational\n🟢")
        return ["Status File Error", "🔴", ed_status()]

File: src/utils/status.go
package utils

const ED_SERVER_URL string = "https://ed-server-status.orerve.net/"

type ED_STATUS struct {
	status  string
	message string
	code    int
	product string
}

func getAppStatus() string {
	return ""
}

func getEDStatus() string {
	jsonData := HTTPGetJSON(ED_SERVER_URL)
	return "ED: " + jsonData["status"].(string)
}

func GetStatus() string {
	return getAppStatus() //+ ", " + getEDStatus()
}
